docs/user-guide/expressions/structs.md

```diff
@@ -0,0 +1,86 @@
+# The Struct datatype
+
+Polars `Sturct`s are an excellent gateway to operations that require multiple columns. For this section, let's start with a `DataFrame` that captures the average rating of a few movies across some states in the U.S.: 
+
+{{code_block('user-guide/expressions/structs','ratings_df',['DataFrame'])}}
+```python exec="on" result="text" session="user-guide/structs"
+--8<-- "python/user-guide/expressions/structs.py:setup"
+--8<-- "python/user-guide/expressions/structs.py:ratings_df"
+```
+
+## Encountering the `Struct` type
+
+A popular operation that will lead to a `Struct` column is the ever so popular `value_counts` function that is commonly used in exploratory data analysis. Checking the number of times a state appears the data will be done as so:
+
+{{code_block('user-guide/expressions/structs','state_value_counts',['value_counts'])}}
+```python exec="on" result="text" session="user-guide/structs"
+--8<-- "python/user-guide/expressions/structs.py:state_value_counts"
+```
+
+Quite unexpected an output, especially if coming from tools that do not have such a data type. We're not in peril though, to get back to a more familiar output, all we need to do is `unnest` the `Struct` column into its constituent columns:
+
+{{code_block('user-guide/expressions/structs','struct_unnest',['unnest'])}}
+```python exec="on" result="text" session="user-guide/structs"
+--8<-- "python/user-guide/expressions/structs.py:struct_unnest"
+```
+
+## Structs as `dict`s
+
+A Polars `Struct` column is similar to a Python `dict`, and is an idiomatic way of working with multiple columns in Polars. Polars will interpret a `dict` sent to the `Series` constructor as a `Struct`:
```

Comment:
I wouldn't make that claim. We use a `dict` in python because we lack a better dtype on the python side. But a struct is not a hashmap.

Reply:
Understood. Will rephrase to add that subtlety.